crosshair/libimpl/builtinslib.py

```diff
@@ -811,9 +828,13 @@ def __round__(self, ndigits=None):
 
 
 class SymbolicInt(AtomicSymbolicValue, SymbolicIntable):
-    def __init__(self, smtvar: Union[str, z3.ExprRef], typ: Type = int):
+    def __init__(self, var: z3.ExprRef, typ: Type = None):
+        typ = int if typ is None else typ
         assert typ == int
-        SymbolicIntable.__init__(self, smtvar, typ)
+        SymbolicIntable.__init__(self, var, typ)
+
+    def __new__(cls, value: int):
+        return cls.from_z3(z3.IntVal(value))
```

Comment:
I'll note that the native python constructors we're emulating can be pretty fancy, e.g. `int("8a1f", 16)`.
What happens if we started with this? (on AtomicSymbolicValue maybe):
```
def __new__(cls, *a):
  return cls._pytype()(*a)
```
This is worse than your solution because it will realize all symbolic state in the arguments. (or at least I think it will; if not, we will need to call `realize()` on the arguments explicitly) But perhaps it's easier to just try to get the behavior right first. We can optimize for preserving symbolics later, where we think we really want it.